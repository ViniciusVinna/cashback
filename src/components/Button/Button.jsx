import React from 'react';
import Ink from 'react-ink';
import PropTypes from 'prop-types';

import { ButtonStyled } from './Button.styles';

const {
  Default,
  Primary,
  Secondary,
} = ButtonStyled;

const Button = ({ children, variation, ...rest }) => {
  if (variation === 'primary') {
    return (
      <Primary type="button" data-test-id="button" {...rest}>
        {children}
        <Ink />
      </Primary>
    );
  }
  if (variation === 'secondary') {
    return (
      <Secondary type="button" data-test-id="button" {...rest}>
        {children}
        <Ink />
      </Secondary>
    );
  }

  return (
    <Default type="button" data-test-id="button">
      {children}
      <Ink />
    </Default>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  variation: PropTypes.string,
};

Button.defaultProps = {
  variation: 'default',
};

export default Button;
