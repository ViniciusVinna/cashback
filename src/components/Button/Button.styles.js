import styled from '@emotion/styled';
import { css } from '@emotion/core';

export const ButtonStyled = {
  Primary: styled('button')(({ theme, disabled }) => css`
    -webkit-tap-highlight-color: transparent;
    background-color: ${disabled ? theme.color.n500 : theme.color.primary};
    border-radius: ${theme.borderRadius.m};
    border: ${theme.borderSize.s} solid ${disabled ? theme.color.n500 : theme.color.primary};
    box-shadow: ${theme.boxShadow.default};
    color: ${disabled ? theme.color.black : theme.color.n100};
    cursor: ${disabled ? 'not-allowed' : 'pointer'};
    font-size: ${theme.textSize.s};
    height: ${theme.iconSize.xxl};
    outline: none;
    overflow: hidden;
    padding: ${theme.space.m} ${theme.space.m};
    position: relative;
    width: 100%;
  `),
  Secondary: styled('button')(({ theme, disabled }) => css`
    -webkit-tap-highlight-color: transparent;
    background-color: ${disabled ? theme.color.n500 : theme.color.secondary};
    border-radius: ${theme.borderRadius.m};
    border: ${theme.borderSize.s} solid ${disabled ? theme.color.n500 : theme.color.secondary};
    box-shadow: ${theme.boxShadow.default};
    color: ${disabled ? theme.color.black : theme.color.n100};
    cursor: ${disabled ? 'not-allowed' : 'pointer'};
    font-size: ${theme.textSize.s};
    height: ${theme.iconSize.xxl};
    outline: none;
    overflow: hidden;
    padding: ${theme.space.m} ${theme.space.m};
    position: relative;
    width: 100%;
  `),
  Default: styled('button')(({ theme, disabled }) => css`
    -webkit-tap-highlight-color: transparent;
    background-color: ${disabled ? theme.color.n500 : theme.color.primary};
    border-radius: ${theme.borderRadius.m};
    border: ${theme.borderSize.s} solid ${disabled ? theme.color.n500 : theme.color.primary};
    box-shadow: ${theme.boxShadow.default};
    color: ${disabled ? theme.color.black : theme.color.n100};
    cursor: ${disabled ? 'not-allowed' : 'pointer'};
    font-size: ${theme.textSize.s};
    height: ${theme.iconSize.xxl};
    outline: none;
    overflow: hidden;
    padding: ${theme.space.m} ${theme.space.m};
    position: relative;
    width: 100%;
  `),
};
