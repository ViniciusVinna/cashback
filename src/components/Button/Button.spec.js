import React from 'react';
import Button from './Button';

describe('Button', () => {
  it('it should render the default button when no props is declared', () => {
    const { container } = render(<Button>Default Button</Button>);

    expect(container).toMatchSnapshot();
  });

  it('it should render the primary button', () => {
    const { container } = render(<Button variation="primary">Primary Button</Button>);

    expect(container).toMatchSnapshot();
  });

  it('it should render the secondary button', () => {
    const { container } = render(<Button variation="secondary">Secondary Button</Button>);

    expect(container).toMatchSnapshot();
  });
});
