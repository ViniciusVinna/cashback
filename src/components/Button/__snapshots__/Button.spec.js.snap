// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it should render the default button when no props is declared 1`] = `
<div>
  <button
    class="css-tdfxln"
    data-test-id="button"
    type="button"
  >
    Default Button
    <canvas
      height="0"
      style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
      width="0"
    />
  </button>
</div>
`;

exports[`Button it should render the primary button 1`] = `
<div>
  <button
    class="css-tdfxln"
    data-test-id="button"
    type="button"
  >
    Primary Button
    <canvas
      height="0"
      style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
      width="0"
    />
  </button>
</div>
`;

exports[`Button it should render the secondary button 1`] = `
<div>
  <button
    class="css-exvprd"
    data-test-id="button"
    type="button"
  >
    Secondary Button
    <canvas
      height="0"
      style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
      width="0"
    />
  </button>
</div>
`;
